### PR TITLE
Fix python codestyle

### PR DESCRIPTION
**Pull request by mathbunnyru**

### CODE REVIEW

**Comment by mathbunnyru** [comment]
Ok, I used flake8 to check all the files, because I'm familiar with it.
So, if you don't mind, I'll add it first in a pre-commit, like you suggested.

Then, if I feel comfortable adding Black, I will also add it, but in a separate review.

Hope this is ok?

**Comment by mathbunnyru** [comment]
https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/
Seems to be good article to start with

**Comment by romainx** [comment]
@mathbunnyru It's Ok to use another formatter. The main idea is to enforce formatting by a hook in order to keep code consistent. I only suggested Black because it has some interesting features and one of the best is summarized by

> Black is opinionated so you don’t have to be.

So please go ahead with flake8 if your more confortable.

Many thanks for your help.
Best.

**Comment by mathbunnyru** [comment]
I tried to look at pre-commit module, but it seems to have a problem - it is still a pre-commit hook.
It means that if user doesn't install it, then nothing will be checked.

And we want to check all the files, server-side - this is the only way to ensure, that everything passes codestyle.

So I added one simple Makefile step, which is included in `lint-all` and `lint-build-test-all` stages.

Please, let me know what you think about this approach.

**Comment by romainx** [comment]
Hello @mathbunnyru.
Sorry for my late answer. 

I think your solution is good to check at server-side before merging a PR if the code is formatted. So let's keep it.

However I still think that the use of [pre-commit](https://pre-commit.com/) is a good solution that can be used in addition to the server side check. The advantage is that the hook is run at each commit and users will not forget to format the file before committing. It will avoid a failure in checks at server side that will lead to unnecessary iterations. My fear is that adding only a server-side check without the pre-commit counterpart will lead to rejected PR.

Maybe I could draft it to add a hook for the hadolint check #1100 ?

**Comment by mathbunnyru** [comment]
Thanks @romainx 👍 
I took a look in change https://github.com/jupyter/docker-stacks/pull/1123 and asked for some changes to make it easier to adapt in the future.
We should merge you pre-commit hook first, and then deal with this review.

**Comment by mathbunnyru** [comment]
`pre-commit run --all-files flake8` seems to pass.
If I make some script ugly, then I get a fail, so everything seems to work.

**Comment by mathbunnyru** [comment]
@romainx @parente could you please take a look?

**Comment by romainx** [comment]
@mathbunnyru Sorry I don't have too much time right now, I will take a look asap.

**Comment by mathbunnyru** [comment]
@romainx any chance you will take a look or should I close the PR?

To be honest, waiting for such a long time makes me sad about creating new PRs and doing some work for this project😔

**Comment by romainx** [comment]
@mathbunnyru sad to hear that 😢. Sorry but I did not take the time to review that. I will try to do it. However I do not have so much time and I'm trying to focus on user facing features / bugs. I though it was a long term internal improvement and postponed this review. 
I will try to do it this weekend.
My intend was not to discourage you for submitting such useful PR. So please forgive my lack of answer and keep contributing to this project 😊.

**Comment by mathbunnyru** [comment]
Thanks, @romainx ☺️

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@ repos:
     rev: 2.0.0
     hooks:
       - id: bashate
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.8.3
+    hooks:
+      - id: flake8
   - repo: local
```
Comment (romainx) [inline .pre-commit-config.yaml:22]: Could you please add the [autopep8 hook](https://github.com/pre-commit/mirrors-autopep8)

```YAML 
 - repo: https://github.com/pre-commit/mirrors-autopep8
    rev: v1.5.4
    hooks:
      - id: autopep8
```

**Review by romainx — CHANGES_REQUESTED** [review]
Hello,

The addition of the [autopep8 hook](https://github.com/pre-commit/mirrors-autopep8) will permit to automatically fix errors.

For example. 

- Add blank lines in `test_pandoc.py` to trigger an error.
- Stage the file for commit.
- Trigger a first run of `pre-commit`.

```bash
$ pre-commit run                                                                              

# [...]
# flake8...................................................................Failed
# - hook id: flake8
# - exit code: 1
# 
# base-notebook/test/test_pandoc.py:17:1: W293 blank line contains whitespace
# base-notebook/test/test_pandoc.py:18:5: E303 too many blank lines (3)
# 
# autopep8.................................................................Failed
# - hook id: autopep8
# - files were modified by this hook
# [...]
```
The formatting issue is detected by `flake8` **and fixed by `autopep8`**.

- Stage the modified file for commit.
- Trigger again the run of `pre-commit`.

```bash
$ pre-commit run                                                                              

flake8...................................................................Passed
autopep8.................................................................Passed
```

All good now!

Could you please perform this change.
After that I will merge it 🚀.

Many thanks for your help and sorry again for the delay 🐌 🐌.

**Comment by parente** [comment]
Thank you @mathbunnyru for the PR and @romainx for taking the time to review it. I appreciate both of your efforts. My availability to contribute here ebbs and flows too, and I think that's OK. We will review and merge PRs that benefit the project, even if not as quickly as some of the larger, more active Jupyter projects.

**Comment by romainx** [comment]
@mathbunnyru I can also merge the PR as is and add `autopep8` through a follow-on PR. Tell me what you prefer.
Thanks

**Comment by mathbunnyru** [comment]
I'm totally OK with both scenarios.
I have some days off till Monday, so I won't be able to add autopep8 this Saturday and Sunday.

**Comment by romainx** [comment]
Ok @mathbunnyru no problem I've waited more than one month before reviewing it, so I can wait a couple of days 😆 
Take the necessary time to do it.

**Comment by mathbunnyru** [comment]
Merge latest master

**Comment by mathbunnyru** [comment]
Added autopep8.

I think everything should be good to go.

**Review by romainx — APPROVED** [review]
Perfect.
Thank you for all the work done 🍰 and for your patience 😊 
Good to go 🚀

**Pull request merged**